### ISSUE TITLE
NodeJS - Unable to increase "grpc.max_send_message_length"

### ISSUE THREAD

**Issue by rkdirect** [grpc/grpc#12012]
### Should this be an issue in the gRPC issue tracker?

Unknown
 
### What version of gRPC and what language are you using?

npm grpc@1.4.1
 
### What operating system (Linux, Windows, …) and version?

Ubuntu 14.04 
 
### What runtime / compiler are you using (e.g. python version or version of gcc)
 
nodejs 6.11
 
### What did you do?

    new grpc.Server({'grpc.max_send_message_length': 8000000,
            'grpc.max_receive_message_length': 1024})
 
### What did you expect to see?

Expected to be able to return large message structures, up to 8m bytes. 
 
### What did you see instead?
 
The limit stays at 4mb.

> "message": "Received message larger than max (6132324 vs. 4194304)",

### Anything else we should know about your project / environment?

If I reduce the value instead to 1024, it does change the setting:

> "message": "Sent message larger than max (6133557 vs. 1024)",

Also of note: setting it this low causes it to crash in the crash handler:

	./node_modules/grpc/src/node/src/server.js:86
	  call.startBatch(error_batch, function(){});
		   ^
	
	Error: startBatch failed
		at Error (native)
		at handleError (./node_modules/grpc/src/node/src/server.js:86:8)
		at ServerUnaryCall.<anonymous> (./node_modules/grpc/src/node/src/server.js:587:5)
		at emitOne (events.js:96:13)
		at ServerUnaryCall.emit (events.js:188:7)
		at ./node_modules/grpc/src/node/src/server.js:500:14
	[nodemon] app crashed - waiting for file changes before starting...

**Comment by murgatroid99** [comment]
It's a little hard to follow exactly what happened here. You say that you're trying to increase the maximum size of messages you can send, but then you change both the max send message size and the max receive message size, and the error you show is referring to a message you received, not one that you sent.

**Comment by rkdirect** [comment]
Oh, snap!
Thanks @murgatroid99 I didn't see that message change from "Sent" to "Received" :(

That's a very interesting piece of news, looks like that may then be originating in endpoint's ESP rather than in my service.

I guess these settings are working as advertised then, I'm just confused about the origination of the error. Thanks for spotting that, sorry to bother you.
